### DISCUSSION TITLE
What will happen to the old M3?

### DISCUSSION THREAD

**Discussion by mtdvlpr** [sircharlo/mmm-refactor#197] [General]
Hey @sircharlo, awesome job creating this refactor! I have yet to test it out myself, but I'm only hearing great things. I believe I saw in the code that you're planning to officially release this refactor somewhere next month? Nice! 

I'm curious what your plan is for the transition period. Will the old M3 remain available for those that want to continue to use some of the features that are not available in the new one? Or will it be like a regular update that's installed automatically (unless disabled in settings)?

Can I be of any assistance during this crucial period? I don't have a lot of time on my hands anymore, but if there are specific questions or problems, I'm happy to help!

I'm very grateful for all the time and effort you have put in this project, old and new. It has made my life and the lives of our other AV brothers so much easier!

**Comment by sircharlo** [comment]
Hi @mtdvlpr!

You got it right! The official release is planned for sometime in October. Right now, I'm focused on ironing out a few minor last-minute bugs and giving everything a final polish.

Regarding the transition, it will be a regular update that's installed automatically. Users who rely on the older features that won’t be available in the refactor are mostly power users anyway. They'll likely either disable auto-updates themselves or reach out via GitHub Discussions.

I want to sincerely express my gratitude for all the help you’ve provided me over the years. You’ve been pivotal in making MMM a valuable tool for congregations worldwide! Right now, things are going pretty smoothly, but I’ll of course keep you on as an admin. If you ever have the time and want to contribute again, your input will always be welcome!

**Comment by mtdvlpr** [comment]
Hey @sircharlo, I've been thinking. I want to contribute, despite my limited time. What I propose is that I clean up the code to make it more maintainable (like I did at the very beginning by splitting the `main.js` file into separate files). Instead of working on one big refactor, I'd like to refactor small bits at a time, by creating some more separate components and utility functions. Slowly chipping away at the 1500 (!) lines of code in the main layout file, for example...

This way, I get to know the new codebase little by little, you get to focus on the features and bugs and the codebase gets more readable and maintainable in the process. What do you think?

**Reply by sircharlo** [reply]
I think that’s a great idea. Refactoring in small, manageable pieces will really help improve the codebase without the pressure of a massive overhaul. Your contributions will make a big difference, and I’m grateful for your help. It’ll be great to have your eyes on the code, and it’ll definitely free me up to focus on features and bugs. Thanks for being willing to take this on! I'm 100% on-board.

**Comment by ux-git** [comment]
@sircharlo will the new version be released within the original GitHub repository? What happens to this refactor repo then?

**Reply by sircharlo** [reply]
The new version will be released in the original repo and the old version should auto-update to the new version seamlessly. When the new version has been released, I will archive this refactor repo.
